advanced-usage/future-mocking.md

```diff
@@ -1,619 +1,513 @@
+
+#### __[C#] Example 5: Construct fake collection and arrange property get to return it__
+
+{{region FutureMocking#SampleCode7Test}}
+
+    public IList<object> FakeCollection()
+    {
+        List<object> resultCollection = new List<object>();
+
+        resultCollection.Add("asd");
+        resultCollection.Add(123);
+        resultCollection.Add(true);
+
+        return resultCollection;
+    }
+    
+    [TestMethod]
+    public void ShouldReturnFakeCollectionForFutureCall()
+    {
+        var fooMocked = Mock.Create<Foo>();
+
+        var expectedCollection = FakeCollection();
+
+        // Arrange
+        Mock.Arrange(() => fooMocked.RealCollection).IgnoreInstance().ReturnsCollection(expectedCollection);
+
+        // Act
+        var actualArrangedCollection = fooMocked.RealCollection;
+        var actualUnArrangedCollection = new Foo().RealCollection;
+
+        // Assert
+        // Asserting for the arranged instance
+        Assert.AreEqual(expectedCollection.Count, actualArrangedCollection.Count);
+        Assert.AreEqual(expectedCollection.FirstOrDefault(), actualArrangedCollection.FirstOrDefault());
+
+        // Asserting for a new unarranged instance
+        Assert.AreEqual(expectedCollection.Count, actualUnArrangedCollection.Count);
+        Assert.AreEqual(expectedCollection.FirstOrDefault(), actualUnArrangedCollection.FirstOrDefault());
+    }
+{{endregion}}
+
+#### __[VB] Example 5: Construct fake collection and arrange property get to return it__
+
+{{region FutureMocking#SampleCode7Test}}
+
+    Public Function FakeCollection() As IList(Of Object)
+        Dim resultCollection As New List(Of Object)()
+
+        resultCollection.Add("asd")
+        resultCollection.Add(123)
+        resultCollection.Add(True)
+
+        Return resultCollection
+    End Function
+
+    <TestMethod>
+    Public Sub ShouldReturnFakeCollectionForFutureCall()
+        Dim fooMocked = Mock.Create(Of Foo)()
+
+        Dim expectedCollection = FakeCollection()
+
+        ' Arrange
+        Mock.Arrange(Function() fooMocked.RealCollection).IgnoreInstance().ReturnsCollection(expectedCollection)
+
+        ' Act
+        Dim actualArrangedCollection = fooMocked.RealCollection
+        Dim actualUnArrangedCollection = New Foo().RealCollection
+
+        ' Assert
+        ' Asserting for the arranged instance
+        Assert.AreEqual(expectedCollection.Count, actualArrangedCollection.Count)
+        Assert.AreEqual(expectedCollection.FirstOrDefault(), actualArrangedCollection.FirstOrDefault())
+
+        ' Asserting for a new unarranged instance
+        Assert.AreEqual(expectedCollection.Count, actualUnArrangedCollection.Count)
+        Assert.AreEqual(expectedCollection.FirstOrDefault(), actualUnArrangedCollection.FirstOrDefault())
+    End Sub
+{{endregion}}
+
+You can see that the test logic is the same as in the previous tests, with the only difference that a collection is being returned this time.
+
+## Mocking the Creation of Objects
```

Comment:
This section should be the first one in this article and the title should be `Mock the Creation of a New Class Object`. The description that follows the section title should also be SEO-friendly. Something like the following "creation of a new class instance in the code under test is typically a great challenge when unit testing". When searching for Mock new class instance/object this article is shown along with the current section and the description. Google shows around 160 characters meaning that the section title and beginning of the description must contain the most valuable of the description.